app/controllers/teachers_controller.rb

```diff
@@ -1,2 +1,31 @@
 class TeachersController < ApplicationController
+  def index
+    @teachers = Teacher.all
+  end
+
+  def new
+    @teacher = Teacher.new
+  end
+
+  def create
+    @teacher = Teacher.new(
+    name: params[:name],
+    bio: params[:bio],
+    certifications: params[:certifications]
+    )
+  end
+
+  def edit
+    @teacher = Teacher.find_by(id: params[:id])
+  end
+
+  def update
+    @teacher = Teacher.find_by(id: params[:id])
+  end
+
+  def destroy
+    @teacher = Teacher.find_by(id: params[:id])
+     @teacher.destroy
+  end
+
```

Comment:
There is one action missing from this controller-- the show action. I'm not sure if I mentioned that show should be included, if I didn't, I apologize. But go ahead and add the show action as well as a show view.